controllers/capabilities/proxyconfigpromote_controller.go

```diff
@@ -176,30 +195,34 @@ func (r *ProxyConfigPromoteReconciler) proxyConfigPromoteReconciler(proxyConfigP
 			}
 			latestStagingVersion = stageElement.ProxyConfig.Version
 
+			// Retrieving latest production version
 			productionElement, err := threescaleAPIClient.GetLatestProxyConfig(productIDStr, "production")
 			if err != nil {
 				reqLogger.Info("Error while finding production version")
 				if !threescaleapi.IsNotFound(err) {
-					//status := r.proxyConfigPromoteStatus(productIDStr, proxyConfigPromote, err, reqLogger, "Failed")
 					statusReconciler := NewProxyConfigPromoteStatusReconciler(r.BaseReconciler, proxyConfigPromote, "Failed", productIDStr, 0, latestStagingVersion, err)
```

Comment:
hey @austincunningham can you think of the reason this would fail (apart from comms interruption) that this would fail? 
I'm starting to think that the error message isn't very clear to the user on what's going on and was wondering if a custom error with clean description on what went wrong would be better here? WDYT?